### PR TITLE
mesa: 11.2.2 -> 12.0.1

### PR DESCRIPTION
**Pull request by rnhmjoj**
###### Things done
- [x] Tested using sandboxing
  ([nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS,
    or option `build-use-chroot` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change (I don't think i'll ever make it)
- [x] Tested execution of all binary files (none)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/mesa/default.nix [hunk 1]
```diff
@@ -124,44 +125,43 @@ stdenv.mkDerivation {
...
 
-  # move gallium-related stuff to $drivers, so $out doesn't depend on LLVM;
-  #   also move libOSMesa to $osmesa, as it's relatively big
-  # ToDo: probably not all .la files are completely fixed, but it shouldn't matter
-  postInstall = with stdenv.lib; ''
+  # TODO:
+  #  move gallium-related stuff to $drivers, so $out doesn't depend on LLVM;
+  #  move libOSMesa to $osmesa, as it's relatively big;
+  #  probably not all .la files are completely fixed, but it shouldn't matter;
+  postInstall = ''
```
Comment (vcunat) [inline pkgs/development/libraries/mesa/default.nix]: This change obscures the meaning of the comments. The first two lines *are done* by the commands below; only the last one should be TODO.

**Comment by vcunat** [comment]
There's a regression: llvm reference is retained from the main output after this PR.

**Comment by rnhmjoj** [comment]
So, there are more libraries that need to be moved to "$drivers/lib"? I can I check which?

**Comment by vcunat** [comment]
For reference checking I use just `grep hash-of-llvm -R ./result`.

**Comment by rnhmjoj** [comment]
I removed the references and fixed the conflict. What's next?

**Comment by vcunat** [comment]
It looks good. I suppose you haven't tried running a system on some of the drivers?

**Comment by vcunat** [comment]
I staged it, but we should test a little before merging to master.

**Comment by rnhmjoj** [comment]
I have been using it on my system for a while: I have played a couple games with openGL 4.x requirement (that I couldn't run before) and programs with graphic acceleration and everything seems fine to me.

**Comment by vcunat** [comment]
Great, thanks!

**Comment by gebner** [comment]
This PR has been accidentally (?) reverted in 9adad8612b082bcbae30c81678a04b79a44079a4, see #17738.

**Comment by shlevy** [comment]
Fixed now in #17957

**Pull request merged**